### DISCUSSION TITLE
Error loading preloads:

### DISCUSSION THREAD

**Discussion by gersondlarosa** [LaunchX-InnovaccionVirtual/CursoIntroPython#147] [Q&A]
Ya cambie todo  re instale el programa cambie variables de entorno jupiter lo instale dede terminal, cambie la versión de jupieter 
y ahora me manda este error

![preload](https://user-images.githubusercontent.com/68626555/153663246-1ec68bf4-2b27-42a2-a25a-cdccd97c70fa.png)

**Comment by AlexString** [comment]
Hola compañero, recuerda que cuando se trata de ejecutar comandos en la terminal se hace así 
`pip install nombreModulo`

Y en un bloque de código de Jupyter Notebook es así:
`!pip install nombreModulo`

Hazme saber si ese era el problema, saludos.

**Reply by gersondlarosa** [reply]
me siguió mandando el mismo error

**Reply by gersondlarosa** [reply]
en la terminal so lo corrió

![coman](https://user-images.githubusercontent.com/68626555/153665628-5c35994a-6528-48b6-9759-b192da525c8a.png)

**Reply by AlexString** [reply]
Ok compañero, si ya en la terminal pudiste instalar el modulo `ipywidgets` ya no es necesario que hagas `pip install` en un bloque de código, estarías instalando lo que ya tienes instalado. Prueba ahora usando el modulo como lo hace la Kata respectiva y si te ejecuta los bloques de código sin error entonces lo has logrado. 

No olvides marcar tu discusión como resuelta si te ha funcionado :)

**Reply by gersondlarosa** [reply]
Hola compañero ya lo corrí en vsc y me mando lo mismo con el codigo de prueba que puso el sherpha

![cod](https://user-images.githubusercontent.com/68626555/153667546-6938c0e0-f27d-4644-9755-9555aad73613.png)
a

**Reply by AlexString** [reply]
Al parecer tu VS Code esta teniendo problemas con empezar el entorno de Jupyter.
Asegurate de tener instalada la extensión de Jupyter en VS Code y si la tienes instalada prueba desinstalando y volviendo a instalar, me pasó algo parecido.

**Reply by gersondlarosa** [reply]
Ya lo cambie y me sale igual 

y el log es este

Jupyter command `jupyter-notebook` not found.

	at t.JupyterConnectionWaiter.rejectStartPromise (c:\Users\linck\.vscode\extensions\ms-toolsai.jupyter-2022.1.1001821375\out\client\extension.js:51:519420)
	at t._complete (c:\Users\linck\.vscode\extensions\ms-toolsai.jupyter-2022.1.1001821375\out\client\extension.js:51:520072)
	at t.n (c:\Users\linck\.vscode\extensions\ms-toolsai.jupyter-2022.1.1001821375\out\client\extension.js:16:16688)
	at t.__tryOrUnsub (c:\Users\linck\.vscode\extensions\ms-toolsai.jupyter-2022.1.1001821375\out\client\extension.js:16:16930)
	at t.complete (c:\Users\linck\.vscode\extensions\ms-toolsai.jupyter-2022.1.1001821375\out\client\extension.js:16:16823)
	at t._complete (c:\Users\linck\.vscode\extensions\ms-toolsai.jupyter-2022.1.1001821375\out\client\extension.js:16:15287)
	at t.complete (c:\Users\linck\.vscode\extensions\ms-toolsai.jupyter-2022.1.1001821375\out\client\extension.js:16:14986)
	at ChildProcess.<anonymous> (c:\Users\linck\.vscode\extensions\ms-toolsai.jupyter-2022.1.1001821375\out\client\extension.js:24:97233)
	at Object.onceWrapper (events.js:422:26)
	at ChildProcess.emit (events.js:327:22)
	at Process.ChildProcess._handle.onexit (internal/child_process.js:277:12)] {
  category: 'jupyterconnection'
}
Info 15:42:50: Cancel pending cells
Info 15:42:50: Cell 0 executed with state Idle

![33](https://user-images.githubusercontent.com/68626555/153674825-7807b23a-9b2c-4cae-89db-e8a03bf06961.png)

**Comment by alejandronjj** [comment]
A mí me da error en la línea de pip install ipywidgets si uso como kernel python 3.10.2 descargado desde la Microsoft Store. En mi caso, tengo instaladas las versiones 3.8.5 y la 3.10.2 descargadas de python.org y ninguna de esas me da problemas.

No debería de darte error incluso si ya tienes instalada ipywidgets. En ese caso la salida solo te dice que el requerimiento ya está cumplido
![image](https://user-images.githubusercontent.com/99060336/153681920-a316cc86-9756-401c-93b0-05188a13426f.png)

En cuanto al error Error loading preloads: Could not find renderer, el señor stack overflow ofrece dos soluciones (pero solo una de ellas aplica en tu caso ya que te da el error en un archivo nuevo) y es la siguiente:

Al parecer, es un error de la última versión de la extensión de jupyter notebook en algunos ordenadores, ve a la pestaña de extensiones y busca jupyter notebook, dale click a la extensión desarrollada por microsoft y le das click a la flecha que está a lado de instalar y seleccionas instalar otra versión.
![scdsd](https://user-images.githubusercontent.com/99060336/153682587-26f8b199-5544-447f-9724-eab17ba9a0d1.jpg)

Los usuarios de stackoverflow recomiendan instalar una versión con al menos un mes de antiguedad para solucionar el error
![scdsds](https://user-images.githubusercontent.com/99060336/153682804-236631cb-ef0b-43f1-b5bb-4376062265cb.jpg)

La pregunta fue hecha hace 6 meses, pero hay algunas respuetas de usuarios hechas en febrero de 2022 que mencionan que está solución sigue funcionando.
https://stackoverflow.com/questions/68736618/error-loading-preloads-could-not-find-renderer
